.claude/skills/cube-tile-tuning/tile_budget.py

```
@@ -0,0 +1,108 @@
+          f"  weights={a.weights} accum={n_accum} dbuf={a.double_buffer}  [{a.platform}]")
+    print(f"  Acc (L0C)   {kb(acc)} / {kb(bud['acc'])}   {verdict(acc, bud['acc'])}"
+          f"   = M*N*{a.bytes_out}*{n_accum}")
+    print(f"  Mat (L1)    {kb(mat)} / {kb(bud['mat'])}   {verdict(mat, bud['mat'])}"
+          f"   = weight {kb(weight_l1)} + act {kb(act_l1)}  (estimate; +10-25% alignment on device)")
+    cl = "OK" if contig >= a.cache_line else f"UNDER ({contig}B < {a.cache_line}B -> MTE2 wastes line)"
+    print(f"  cache-line  K*bytes_in = {contig} B   {cl}")
+
+    # Headroom / suggestions.
+    print("  ---")
+    if acc > bud["acc"]:
+        print(f"  Acc is the wall: drop N or M, or split accumulators.")
```

Ruff (0.15.18)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)